src/main/java/org/sonar/ucfg/Instruction.java

```diff
@@ -54,6 +55,51 @@ private Terminator(InstructionType type, @Nullable LocationInFile locationInFile
     }
   }
 
+  public static class NewObject extends Instruction {
+    private final Expression.Variable lhs;
+    private final String instanceType;
+
+    private final int hash;
+    public NewObject(LocationInFile locationInFile, Expression.Variable lhs, String instanceType) {
+      super(InstructionType.NEW, locationInFile);
+      this.lhs = lhs;
+      this.instanceType = instanceType;
+      this.hash = Objects.hash(lhs, instanceType, locationInFile);
```

Comment:
Just wondering now, but why not include the `InstructionType` into the hash as well? (same comment for other instructions.)

Reply:
should be done in another ticket